src/Pact/Types/Type.hs

```diff
@@ -121,16 +135,20 @@ tyList = "list"
 tyObject = "object"
 tyValue = "value"
 tyKeySet = "keyset"
+tyGuard = "guard"
 tyTable = "table"
 
 instance Show PrimType where
-  show TyInteger = unpack tyInteger
-  show TyDecimal = unpack tyDecimal
-  show TyTime = unpack tyTime
-  show TyBool = unpack tyBool
-  show TyString = unpack tyString
-  show TyValue = unpack tyValue
-  show TyKeySet = unpack tyKeySet
+  show t = unpack $ case t of
+    TyInteger -> tyInteger
+    TyDecimal -> tyDecimal
+    TyTime -> tyTime
+    TyBool -> tyBool
+    TyString -> tyString
+    TyValue -> tyValue
+    TyGuard tg -> case tg of
+      Just GTyKeySet -> tyKeySet
+      _ -> tyGuard
```

Comment:
so `guard` is the keyword for all guard types except keysets. will that always be the case?

Reply:
Yeah this was something I was hesitating to bring up as well. the `Just GTyKeySet` pattern smells a bit